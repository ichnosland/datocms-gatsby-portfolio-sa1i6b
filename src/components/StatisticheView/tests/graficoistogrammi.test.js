import React from 'react';
import { shallow } from 'enzyme';
import { VerticalBarSeries } from 'react-vis';

import GraficoMedieCompito from '../GraficoIstogrammi';


const mockProps = {
  titoliGrafici: [{
    key: 1,
    titolo: 'titolo 1',
  }, {
    key: 2,
    titolo: 'titolo 2',
  }],
  mediaNazionale: [{
    y: 12,
    x: '#1',
    titolo: 'titolo 1',
  }, {
    y: 22,
    x: '#2',
    titolo: 'titolo 2',
  }],
  mediaClasse: [{
    y: 6,
    x: '#1',
    titolo: 'titolo 1',
  }, {
    y: 5,
    x: '#2',
    titolo: 'titolo 2',
  }],
  mediaStudente: [{
    y: 3,
    x: '#1',
    titolo: 'titolo 1',
  }, {
    y: 2,
    x: '#2',
    titolo: 'titolo 2',
  }],
  onDidascaliaSet: () => { },
  onDidascaliaReset: () => { },
  didascalia: {
    tipologiaGrafico: 'medieCompito',
    display: false,
    tipologia: 'tipologia',
    y: 1,
    x: '#1',
    titolo: 'titolo',
    campioni: 1,
  },
};

describe('<GraficoMedieCompito />', () => {
  it('Mostro la condizione iniziale', () => {
    const renderedComponent = shallow(
      <GraficoMedieCompito
        {...mockProps}
      />
    );

    expect(renderedComponent).toMatchSnapshot();
  });

  it('Mostro la condizione se hint Ã¨ visibile', () => {
    const props = {
      ...mockProps,
      didascalia: {
        ...mockProps.didascalia,
        display: true,
      },
    };
    const renderedComponent = shallow(
      <GraficoMedieCompito
        {...props}
      />
    );

    expect(renderedComponent).toMatchSnapshot();
  });


  it('setDidascalia deve chiamare onDidascaliaSet', () => {
    const props = {
      ...mockProps,
      onDidascaliaSet: jest.fn(),
    };
    const renderedComponent = shallow(
      <GraficoMedieCompito
        {...props}
      />
    );

    expect(props.onDidascaliaSet).not.toHaveBeenCalled();
    const instance = renderedComponent.instance();
    instance.setDidascalia({
      titolo: 'titolo',
      tipologia: 'tipologia',
      y: 2,
      x: 3,
      campioni: 4,
    });
    expect(props.onDidascaliaSet).toHaveBeenCalledWith({
      display: true,
      tipologiaGrafico: 'medieCompito',
      titolo: 'titolo',
      tipologia: 'tipologia',
      y: 2,
      x: 3,
      campioni: 4,
    });
  });


  it('VerticalBarSeries(0) onValueMouseOver deve chiamare onDidascaliaSet', () => {
    const props = {
      ...mockProps,
      onDidascaliaSet: jest.fn(),
    };
    const renderedComponent = shallow(
      <GraficoMedieCompito
        {...props}
      />
    );

    const instance = renderedComponent.instance();
    const spy = jest.spyOn(instance, 'setDidascalia');

    renderedComponent.find(VerticalBarSeries).at(0).props().onValueMouseOver({
      titolo: 'titolo1',
    });
    expect(spy).toHaveBeenCalledWith({ titolo: 'titolo1' });

    renderedComponent.find(VerticalBarSeries).at(1).props().onValueMouseOver({
      titolo: 'titolo2',
    });
    expect(spy).toHaveBeenCalledWith({ titolo: 'titolo2' });

    renderedComponent.find(VerticalBarSeries).at(2).props().onValueMouseOver({
      titolo: 'titolo2',
    });
    expect(spy).toHaveBeenCalledWith({ titolo: 'titolo2' });
  });
});
