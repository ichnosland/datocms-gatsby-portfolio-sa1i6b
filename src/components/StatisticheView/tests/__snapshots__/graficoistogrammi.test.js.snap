// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GraficoMedieCompito /> Mostro la condizione iniziale 1`] = `
<StatsElements__ChartBox
  className="ChartBox istogrammi"
  legenda={true}
>
  <StatsElements__LegendaChartBox>
    <div>
      <StatsElements__SwatchLabel>
        Media di classe
      </StatsElements__SwatchLabel>
      <StatsElements__Swatch
        bgColor="#FFC155"
      />
    </div>
    <div>
      <StatsElements__SwatchLabel>
        Media nazionale
      </StatsElements__SwatchLabel>
      <StatsElements__Swatch
        bgColor="#0F7CFF"
      />
    </div>
    <div>
      <StatsElements__SwatchLabel>
        Punteggio studente
      </StatsElements__SwatchLabel>
      <StatsElements__Swatch
        bgColor="#17B317"
      />
    </div>
  </StatsElements__LegendaChartBox>
  <Div
    className="grafico"
    height="300px"
    width="100%"
  >
    <FlexibleXYPlot
      animation={true}
      colorType="literal"
      xType="ordinal"
      yDomain={
        Array [
          0,
          10,
        ]
      }
    >
      <VerticalGridLines
        attr="x"
        direction="vertical"
      />
      <HorizontalGridLines
        attr="y"
        direction="horizontal"
      />
      <XAxis
        attr="x"
        attrAxis="y"
        orientation="bottom"
      />
      <YAxis
        attr="y"
        attrAxis="x"
        orientation="left"
      />
      <VerticalBarSeries
        className=""
        data={
          Array [
            Object {
              "titolo": "titolo 1",
              "x": "#1",
              "y": 6,
            },
            Object {
              "titolo": "titolo 2",
              "x": "#2",
              "y": 5,
            },
          ]
        }
        onValueMouseOut={[Function]}
        onValueMouseOver={[Function]}
        stack={false}
        style={Object {}}
      />
      <VerticalBarSeries
        className=""
        data={
          Array [
            Object {
              "titolo": "titolo 1",
              "x": "#1",
              "y": 12,
            },
            Object {
              "titolo": "titolo 2",
              "x": "#2",
              "y": 22,
            },
          ]
        }
        onValueMouseOut={[Function]}
        onValueMouseOver={[Function]}
        stack={false}
        style={Object {}}
      />
      <VerticalBarSeries
        className=""
        data={
          Array [
            Object {
              "titolo": "titolo 1",
              "x": "#1",
              "y": 3,
            },
            Object {
              "titolo": "titolo 2",
              "x": "#2",
              "y": 2,
            },
          ]
        }
        onValueMouseOut={[Function]}
        onValueMouseOver={[Function]}
        stack={false}
        style={Object {}}
      />
    </FlexibleXYPlot>
  </Div>
  <FlexBox
    className="titoli-grafico"
    justifyContent="space-between"
  >
    <Div
      padding="0 30px 10px"
    >
      <StatsElements__SwatchLabel
        key="titolo_grafico_1"
      >
        1
        . 
        titolo 1
      </StatsElements__SwatchLabel>
      <StatsElements__SwatchLabel
        key="titolo_grafico_2"
      >
        2
        . 
        titolo 2
      </StatsElements__SwatchLabel>
    </Div>
  </FlexBox>
</StatsElements__ChartBox>
`;

exports[`<GraficoMedieCompito /> Mostro la condizione se hint Ã¨ visibile 1`] = `
<StatsElements__ChartBox
  className="ChartBox istogrammi"
  legenda={true}
>
  <StatsElements__LegendaChartBox>
    <div>
      <StatsElements__SwatchLabel>
        Media di classe
      </StatsElements__SwatchLabel>
      <StatsElements__Swatch
        bgColor="#FFC155"
      />
    </div>
    <div>
      <StatsElements__SwatchLabel>
        Media nazionale
      </StatsElements__SwatchLabel>
      <StatsElements__Swatch
        bgColor="#0F7CFF"
      />
    </div>
    <div>
      <StatsElements__SwatchLabel>
        Punteggio studente
      </StatsElements__SwatchLabel>
      <StatsElements__Swatch
        bgColor="#17B317"
      />
    </div>
  </StatsElements__LegendaChartBox>
  <Div
    className="grafico"
    height="300px"
    width="100%"
  >
    <FlexibleXYPlot
      animation={true}
      colorType="literal"
      xType="ordinal"
      yDomain={
        Array [
          0,
          10,
        ]
      }
    >
      <VerticalGridLines
        attr="x"
        direction="vertical"
      />
      <HorizontalGridLines
        attr="y"
        direction="horizontal"
      />
      <XAxis
        attr="x"
        attrAxis="y"
        orientation="bottom"
      />
      <YAxis
        attr="y"
        attrAxis="x"
        orientation="left"
      />
      <Hint
        align={
          Object {
            "horizontal": "auto",
            "vertical": "auto",
          }
        }
        format={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.6)",
            "borderRadius": "6px",
            "color": "#fff",
            "fontSize": "14px",
            "padding": "6px 10px",
          }
        }
        value={
          Object {
            "campioni": 1,
            "display": true,
            "tipologia": "tipologia",
            "tipologiaGrafico": "medieCompito",
            "titolo": "titolo",
            "x": "#1",
            "y": 1,
          }
        }
      >
        <div>
          <p>
            titolo
          </p>
          <p>
            tipologia
            : 
            1
          </p>
          <p>
            Numero di campioni: 
            1
          </p>
        </div>
      </Hint>
      <VerticalBarSeries
        className=""
        data={
          Array [
            Object {
              "titolo": "titolo 1",
              "x": "#1",
              "y": 6,
            },
            Object {
              "titolo": "titolo 2",
              "x": "#2",
              "y": 5,
            },
          ]
        }
        onValueMouseOut={[Function]}
        onValueMouseOver={[Function]}
        stack={false}
        style={Object {}}
      />
      <VerticalBarSeries
        className=""
        data={
          Array [
            Object {
              "titolo": "titolo 1",
              "x": "#1",
              "y": 12,
            },
            Object {
              "titolo": "titolo 2",
              "x": "#2",
              "y": 22,
            },
          ]
        }
        onValueMouseOut={[Function]}
        onValueMouseOver={[Function]}
        stack={false}
        style={Object {}}
      />
      <VerticalBarSeries
        className=""
        data={
          Array [
            Object {
              "titolo": "titolo 1",
              "x": "#1",
              "y": 3,
            },
            Object {
              "titolo": "titolo 2",
              "x": "#2",
              "y": 2,
            },
          ]
        }
        onValueMouseOut={[Function]}
        onValueMouseOver={[Function]}
        stack={false}
        style={Object {}}
      />
    </FlexibleXYPlot>
  </Div>
  <FlexBox
    className="titoli-grafico"
    justifyContent="space-between"
  >
    <Div
      padding="0 30px 10px"
    >
      <StatsElements__SwatchLabel
        key="titolo_grafico_1"
      >
        1
        . 
        titolo 1
      </StatsElements__SwatchLabel>
      <StatsElements__SwatchLabel
        key="titolo_grafico_2"
      >
        2
        . 
        titolo 2
      </StatsElements__SwatchLabel>
    </Div>
  </FlexBox>
</StatsElements__ChartBox>
`;
